sql/core/src/main/scala/org/apache/spark/sql/execution/script/ScriptTransformationExec.scala

```diff
@@ -0,0 +1,337 @@
+    stderrBuffer: CircularBuffer,
+    taskContext: TaskContext,
+    conf: Configuration
+  ) extends Thread("Thread-ScriptTransformation-Feed") with Logging with Serializable {
+
+  setDaemon(true)
+
+  @volatile protected var _exception: Throwable = null
+
+  protected val lineDelimiter = ioschema.inputRowFormatMap("TOK_TABLEROWFORMATLINES")
+  protected val fieldDelimiter = ioschema.inputRowFormatMap("TOK_TABLEROWFORMATFIELD")
+
+  /** Contains the exception thrown while writing the parent iterator to the external process. */
+  def exception: Option[Throwable] = Option(_exception)
+
+  protected def init(): Unit = {
+    TaskContext.setTaskContext(taskContext)
+  }
+
+  protected def processRow(row: InternalRow, numColumns: Int): Unit = {
+    val data = if (numColumns == 0) {
+      lineDelimiter
+    } else {
+      val sb = new StringBuilder
+      sb.append(row.get(0, inputSchema.head))
+      var i = 1
+      while (i < numColumns) {
+        sb.append(fieldDelimiter)
+        val columnType = inputSchema(i)
+        val fieldValue = row.get(i, columnType)
+        val fieldStringValue = columnType match {
+          case _: DateType =>
+            DateTimeUtils.dateToString(fieldValue.asInstanceOf[SQLDate])
+          case _: TimestampType =>
+            DateTimeUtils.timestampToString(fieldValue.asInstanceOf[SQLTimestamp])
+          case _ =>
+            fieldValue.toString
+        }
+        sb.append(fieldStringValue)
+        i += 1
+      }
+      sb.append(lineDelimiter)
+      sb.toString()
+    }
+    outputStream.write(data.getBytes(StandardCharsets.UTF_8))
+  }
+
+  override def run(): Unit = Utils.logUncaughtExceptions {
+    init()
+
+    // We can't use Utils.tryWithSafeFinally here because we also need a `catch` block, so
+    // let's use a variable to record whether the `finally` block was hit due to an exception
+    var threwException: Boolean = true
+    val numColumns = inputSchema.length
+    try {
+      iter.map(outputProjection).foreach(row => processRow(row, numColumns))
+      threwException = false
+    } catch {
+      case t: Throwable =>
+        // An error occurred while writing input, so kill the child process. According to the
+        // Javadoc this call will not throw an exception:
+        _exception = t
+        proc.destroy()
+        throw t
+    } finally {
+      try {
+        Utils.tryLogNonFatalError(outputStream.close())
+        if (proc.waitFor() != 0) {
+          logError(stderrBuffer.toString) // log the stderr circular buffer
+        }
+      } catch {
+        case NonFatal(exceptionFromFinallyBlock) =>
+          if (!threwException) {
+            throw exceptionFromFinallyBlock
+          } else {
+            log.error("Exception in finally block", exceptionFromFinallyBlock)
+          }
+      }
+    }
+  }
+}
+
+private[sql]
+object ScriptTransformIOSchema {
+  def apply(input: ScriptInputOutputSchema): ScriptTransformIOSchema = {
+    new ScriptTransformIOSchema(
```

Comment:
```Scala
case class ScriptInputOutputSchema(
    inputRowFormat: Seq[(String, String)],
    outputRowFormat: Seq[(String, String)],
    inputSerdeClass: Option[String],
    outputSerdeClass: Option[String],
    inputSerdeProps: Seq[(String, String)],
    outputSerdeProps: Seq[(String, String)],
    recordReaderClass: Option[String],
    recordWriterClass: Option[String],
    schemaLess: Boolean)
```

Except `inputRowFormat `, `outputRowFormat ` and  `schemaLess `, we ignore all the other fields. I think we should not silently ignore them. For example, we do not respect any user-specified conf values of `hive.script.recordreader` and `hive.script.recordwriter`. Thus, could we issue an exception when users set them?